### PR TITLE
KTOR-3217 Add support for the new native memory model

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-3217](https://youtrack.jetbrains.com/issue/KTOR-3217)

### CODE REVIEW

File: ktor-network/nix/src/io/ktor/network/selector/SelectUtils.kt [hunk 1]
```diff
@@ -95,7 +95,7 @@ internal class SelectorHelper {
             processSelectedEvents(watchSet, closeSet, completed, readSet, writeSet, errorSet)
         }
 
-        val exception = CancellationException("Selector closed").freeze()
+        val exception = CancellationException("Selector closed").apply { freeze() }
```
Comment (rsinukov) [inline ktor-network/nix/src/io/ktor/network/selector/SelectUtils.kt:98]: should it be `makeShared`?

**Review by rsinukov — APPROVED** [review]
LGTM

**Comment by rsinukov** [comment]
Please check CI build

**Review by olme04 — COMMENTED** [review]
Minor: we can use check for experimental MM in [shared collections](https://github.com/ktorio/ktor/blob/4a4f0cd584c948a24c79c2809d521e79c0c93a62/ktor-utils/common/src/io/ktor/util/collections/CollectionUtils.kt#L10), to use simple maps instead of concurrentMap based on atomics

**Comment by e5l** [comment]
@olme04, nice observation. Will replace! Thank you

**Comment by olme04** [comment]
BTW, `IS_PLATFORM_BIG_ENDIAN` seems can be replaced by [Platform.isLittleEndian](https://github.com/JetBrains/kotlin/blob/master/kotlin-native/runtime/src/main/kotlin/kotlin/native/Platform.kt#L63) (with inverting of check of course)

File: ktor-network/nix/src/io/ktor/network/selector/SelectUtils.kt (hunk 1 above)
Comment (e5l) [reply ktor-network/nix/src/io/ktor/network/selector/SelectUtils.kt:98]: Thanks, fixed

**Pull request merged**